src/hipscat_import/catalog/resume_plan.py

```diff
@@ -0,0 +1,224 @@
+                ]
+                self.split_keys = [
+                    (key, file)
+                    for (key, file) in self.split_keys
+                    if key not in split_keys
+                ]
+            ## We don't pre-gather the plan for the reducing keys.
+            ## It requires the full destination pixel map.
+            step_progress.update(1)
+
+    def read_histogram(self, highest_healpix_order):
+        """Read a numpy array at the indicated directory.
+        Otherwise, return histogram of appropriate shape."""
+        file_name = file_io.append_paths_to_pointer(
+            self.tmp_path, self.HISTOGRAM_BINARY_FILE
+        )
+        if file_io.does_file_or_directory_exist(file_name):
+            with open(file_name, "rb") as file_handle:
+                return frombuffer(file_handle.read(), dtype=np.int64)
+        return pixel_math.empty_histogram(highest_healpix_order)
+
+    def mark_mapping_done(self, mapping_key: str, histogram):
+        """Add mapping key to done list and update raw histogram"""
+        self._write_log_key(self.MAPPING_START_LOG_FILE, mapping_key)
+        file_name = file_io.append_paths_to_pointer(
+            self.tmp_path, self.HISTOGRAM_BINARY_FILE
+        )
+        with open(file_name, "wb+") as file_handle:
+            file_handle.write(histogram.data)
+        self._write_log_key(self.MAPPING_DONE_LOG_FILE, mapping_key)
+
+    def is_mapping_done(self) -> bool:
+        """Are there files left to map?"""
+        return self._done_file_exists(self.MAPPING_DONE_FILE)
+
+    def set_mapping_done(self):
+        """All files are done mapping."""
+        self._touch_done_file(self.MAPPING_DONE_FILE)
+
+    def mark_splitting_done(self, splitting_key: str):
+        """Add splitting key to done list"""
+        self._write_log_key(self.SPLITTING_LOG_FILE, splitting_key)
+
+    def is_splitting_done(self) -> bool:
+        """Are there files left to split?"""
+        return self._done_file_exists(self.SPLITTING_DONE_FILE)
+
+    def set_splitting_done(self):
+        """All files are done splitting."""
+        self._touch_done_file(self.SPLITTING_DONE_FILE)
+
+    def get_reduce_items(self, destination_pixel_map):
+        """Fetch a triple for each partition to reduce.
+
+        Triple contains:
+
+        - destination pixel (healpix pixel with both order and pixel)
+        - source pixels (list of pixels at mapping order)
+        - reduce key (string of destination order+pixel)
+        """
+        reduced_keys = set(self._read_log_keys(self.REDUCING_LOG_FILE))
+        reduce_items = [
+            (hp_pixel, source_pixels, f"{hp_pixel.order}_{hp_pixel.pixel}")
+            for hp_pixel, source_pixels in destination_pixel_map.items()
+        ]
+        reduce_items = [
+            (hp_pixel, source_pixels, key)
+            for (hp_pixel, source_pixels, key) in reduce_items
+            if key not in reduced_keys
+        ]
```

Comment:
This sets the same variable twice? I'm guessing the first time is the old version